core/middlewares/with-routes.ts

```diff
@@ -251,20 +269,21 @@ export const withRoutes: MiddlewareFactory = () => {
 
     if (route?.redirect) {
       switch (route.redirect.to.__typename) {
-        case 'ManualRedirect': {
-          // For manual redirects, redirect to the full URL to handle cases
-          // where the destination URL might be external to the site
-          return NextResponse.redirect(route.redirect.toUrl, redirectConfig);
+        case 'BlogPostRedirect':
+        case 'BrandRedirect':
+        case 'CategoryRedirect':
+        case 'PageRedirect':
+        case 'ProductRedirect': {
+          // For dynamic redirects, assume an internal redirect and construct the URL from the path
+          const redirectUrl = new URL(route.redirect.to.path, request.url);
+
+          return NextResponse.redirect(redirectUrl, redirectConfig);
```

Comment:
Hey @willPrattUPL I found a bug in the previous logic as it was rewriting, not redirecting. Can you test y'all redirects from this branch, specifically that 404 that had `node` return `null`?

Reply:
I manually tested this with your 404 response, but added a `path` node to the "response". It seemed to work on my end but had to push a change to make sure trailing slashes were preserved correctly.

Reply:
Hey, sorry Chance. We are extremely busy right now but I promise I'll get to this this week @chanceaclark